<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Using the ScPCAr package to access Single-cell Pediatric Cancer Atlas data • ScPCAr</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Using the ScPCAr package to access Single-cell Pediatric Cancer Atlas data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ScPCAr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/ScPCAr.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Using the ScPCAr package to access Single-cell Pediatric Cancer Atlas data</h1>
            
      

      <div class="d-none name"><code>ScPCAr.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The ScPCAr package provides an interface to interact with the
Single-cell Pediatric Cancer Atlas (ScPCA) Portal API. This vignette
demonstrates the basic workflow for discovering, downloading, and
working with ScPCA data.</p>
<p>This vignette covers the following tasks for interacting with the
ScPCA Portal:</p>
<ol style="list-style-type: decimal">
<li>Listing available projects</li>
<li>Selecting a project and exploring its samples</li>
<li>Obtaining an authentication token</li>
<li>Downloading data for individual samples</li>
<li>Loading the data into R</li>
</ol>
</div>
<div class="section level2">
<h2 id="installing-the-scpcar-package">Installing the ScPCAr Package<a class="anchor" aria-label="anchor" href="#installing-the-scpcar-package"></a>
</h2>
<p>The ScPCAr package is currently available via GitHub. You can install
the latest version using the <code>remotes</code> package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install remotes if needed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"remotes"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Install ScPCAr from GitHub</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"Alexslemonade/ScPCAr"</span><span class="op">)</span></span></code></pre></div>
<p>You can then load the <code>ScPCAr</code> package. Note that in this
vignette we will use <code>ScPCAr::</code> when calling functions from
the package, both for clarity and to avoid any possible namespace
conflicts. We will also load the <code>SingleCellExperiment</code>
package for later analysis.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://alexslemonade.github.io/ScPCAr/">ScPCAr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="exploring-available-projects">Exploring available projects<a class="anchor" aria-label="anchor" href="#exploring-available-projects"></a>
</h2>
<div class="section level3">
<h3 id="listing-all-projects">Listing all projects<a class="anchor" aria-label="anchor" href="#listing-all-projects"></a>
</h3>
<p>First, let’s see what projects are available in the ScPCA Portal:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get a data frame of all projects</span></span>
<span><span class="va">projects</span> <span class="op">&lt;-</span> <span class="fu">ScPCAr</span><span class="fu">::</span><span class="fu"><a href="../reference/scpca_projects.html">scpca_projects</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># print out a portion of the data frame</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">projects</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## # A tibble: 6 × 23</span></span>
<span><span class="co">##   scpca_project_id sample_count title    pi_name abstract additional_restricti…¹</span></span>
<span><span class="co">##   &lt;chr&gt;                   &lt;int&gt; &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;                 </span></span>
<span><span class="co">## 1 SCPCP000001                23 Single … green_… Pediatr… Research or academic …</span></span>
<span><span class="co">## 2 SCPCP000002                26 Single … green_… Pediatr… Research or academic …</span></span>
<span><span class="co">## 3 SCPCP000003                59 Single … teache… Early T… Research or academic …</span></span>
<span><span class="co">## 4 SCPCP000006                45 Single … murphy… Wilms t… Research or academic …</span></span>
<span><span class="co">## 5 SCPCP000007                30 Single-… gawad   Bulk ge… Research or academic …</span></span>
<span><span class="co">## 6 SCPCP000008               104 Single-… mullig… Acute l… Research or academic …</span></span>
<span><span class="co">## # ℹ abbreviated name: ¹​additional_restrictions</span></span>
<span><span class="co">## # ℹ 17 more variables: created_at &lt;dttm&gt;, downloadable_sample_count &lt;int&gt;,</span></span>
<span><span class="co">## #   has_bulk_rna_seq &lt;lgl&gt;, has_cite_seq_data &lt;lgl&gt;,</span></span>
<span><span class="co">## #   has_multiplexed_data &lt;lgl&gt;, has_single_cell_data &lt;lgl&gt;,</span></span>
<span><span class="co">## #   has_spatial_data &lt;lgl&gt;, human_readable_pi_name &lt;chr&gt;,</span></span>
<span><span class="co">## #   includes_anndata &lt;lgl&gt;, includes_cell_lines &lt;lgl&gt;,</span></span>
<span><span class="co">## #   includes_merged_anndata &lt;lgl&gt;, includes_merged_sce &lt;lgl&gt;, …</span></span></code></pre>
<p>The <code><a href="../reference/scpca_projects.html">scpca_projects()</a></code> function returns a data frame with
basic project metadata. By default, it returns a simplified version with
list columns removed for easier viewing. You can see the full structure
with additional data such as a list of diagnoses, data types, external
accession numbers, etc., by setting <code>simplify = FALSE</code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the full project information including list columns</span></span>
<span><span class="va">projects_full</span> <span class="op">&lt;-</span> <span class="fu">ScPCAr</span><span class="fu">::</span><span class="fu"><a href="../reference/scpca_projects.html">scpca_projects</a></span><span class="op">(</span>simplify <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View the structure of the full data frame</span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="va">projects_full</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Rows: 23</span></span>
<span><span class="co">## Columns: 37</span></span>
<span><span class="co">## $ scpca_project_id          &lt;chr&gt; "SCPCP000001", "SCPCP000002", "SCPCP000003",…</span></span>
<span><span class="co">## $ sample_count              &lt;int&gt; 23, 26, 59, 45, 30, 104, 39, 42, 11, 33, 10,…</span></span>
<span><span class="co">## $ title                     &lt;chr&gt; "Single cell RNA sequencing of pediatric hig…</span></span>
<span><span class="co">## $ pi_name                   &lt;chr&gt; "green_mulcahy_levy", "green_mulcahy_levy", …</span></span>
<span><span class="co">## $ abstract                  &lt;chr&gt; "Pediatric brain tumors are now the most com…</span></span>
<span><span class="co">## $ additional_metadata_keys  &lt;list&gt; &lt;"development_stage_ontology_term_id", "dis…</span></span>
<span><span class="co">## $ additional_restrictions   &lt;chr&gt; "Research or academic purposes only", "Resea…</span></span>
<span><span class="co">## $ computed_files            &lt;list&gt; [&lt;data.frame[5 x 17]&gt;], [&lt;data.frame[5 x 17…</span></span>
<span><span class="co">## $ contacts                  &lt;list&gt; [&lt;data.frame[1 x 2]&gt;], [&lt;data.frame[1 x 2]&gt;…</span></span>
<span><span class="co">## $ created_at                &lt;dttm&gt; 2025-08-26, 2025-08-26, 2025-08-26, 2025-08…</span></span>
<span><span class="co">## $ diagnoses_counts          &lt;df[,56]&gt; &lt;data.frame[23 x 56]&gt;</span></span>
<span><span class="co">## $ diagnoses                 &lt;list&gt; &lt;"Anaplastic astrocytoma", "Anaplastic g…</span></span>
<span><span class="co">## $ disease_timings           &lt;list&gt; &lt;"Metastatic recurrence of anaplastic pleom…</span></span>
<span><span class="co">## $ downloadable_sample_count &lt;int&gt; 23, 26, 59, 43, 30, 104, 38, 42, 11, 33, 10…</span></span>
<span><span class="co">## $ external_accessions       &lt;list&gt; [&lt;data.frame[4 x 3]&gt;], [&lt;data.frame[0 x 0]&gt;]…</span></span>
<span><span class="co">## $ has_bulk_rna_seq          &lt;lgl&gt; TRUE, TRUE, TRUE, TRUE, FALSE, FALSE, TRUE,…</span></span>
<span><span class="co">## $ has_cite_seq_data         &lt;lgl&gt; FALSE, FALSE, TRUE, FALSE, TRUE, TRUE, FALSE…</span></span>
<span><span class="co">## $ has_multiplexed_data      &lt;lgl&gt; FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, TR…</span></span>
<span><span class="co">## $ has_single_cell_data      &lt;lgl&gt; TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TR…</span></span>
<span><span class="co">## $ has_spatial_data          &lt;lgl&gt; FALSE, FALSE, FALSE, TRUE, FALSE, FALSE, FAL…</span></span>
<span><span class="co">## $ human_readable_pi_name    &lt;chr&gt; "Green/Mulcahy Levy", "Green/Mulcahy Levy", …</span></span>
<span><span class="co">## $ includes_anndata          &lt;lgl&gt; TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TR…</span></span>
<span><span class="co">## $ includes_cell_lines       &lt;lgl&gt; FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FA…</span></span>
<span><span class="co">## $ includes_merged_anndata   &lt;lgl&gt; TRUE, TRUE, TRUE, TRUE, TRUE, FALSE, FALSE, …</span></span>
<span><span class="co">## $ includes_merged_sce       &lt;lgl&gt; TRUE, TRUE, TRUE, TRUE, TRUE, FALSE, FALSE, …</span></span>
<span><span class="co">## $ includes_xenografts       &lt;lgl&gt; FALSE, FALSE, TRUE, FALSE, FALSE, FALSE, FAL…</span></span>
<span><span class="co">## $ metadata_dataset_id       &lt;lgl&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, …</span></span>
<span><span class="co">## $ modalities                &lt;list&gt; &lt;"SINGLE_CELL", "BULK_RNA_SEQ"&gt;, &lt;"SINGLE_CE…</span></span>
<span><span class="co">## $ multiplexed_sample_count  &lt;int&gt; 0, 0, 0, 0, 0, 0, 34, 0, 0, 0, 0, 0, 0, 0, …</span></span>
<span><span class="co">## $ organisms                 &lt;list&gt; "Homo sapiens", "Homo sapiens", "Homo sapien…</span></span>
<span><span class="co">## $ publications              &lt;list&gt; [&lt;data.frame[1 x 3]&gt;], [&lt;data.frame[0 x 0]&gt;…</span></span>
<span><span class="co">## $ samples                   &lt;list&gt; &lt;"SCPCS000001", "SCPCS000002", "SCPCS000003…</span></span>
<span><span class="co">## $ seq_units                 &lt;list&gt; "cell", "cell", "cell", &lt;"nucleus", "spot"&gt;…</span></span>
<span><span class="co">## $ summaries                 &lt;list&gt; [&lt;data.frame[7 x 5]&gt;], [&lt;data.frame[5 x 5]&gt;…</span></span>
<span><span class="co">## $ technologies              &lt;list&gt; "10Xv3", "10Xv3", "10Xv3", &lt;"10Xv3.1", "vis…</span></span>
<span><span class="co">## $ unavailable_samples_count &lt;int&gt; 0, 0, 0, 2, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0…</span></span>
<span><span class="co">## $ updated_at                &lt;dttm&gt; 2025-08-26, 2025-08-26, 2025-08-26, 2025-08-…</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="getting-detailed-project-information">Getting detailed project information<a class="anchor" aria-label="anchor" href="#getting-detailed-project-information"></a>
</h3>
<p>Now let’s get more detailed information about the samples in a
selected project. We will use the first project,
<code>SCPCP000001</code>, as an example. According to the project info
above, these samples are from a study of pediatric high-grade gliomas.
We can get more detailed information about this project using its
<code>project_id</code>, and here we will set
<code>simplifyVector = TRUE</code> to convert simple lists into vectors
where possible.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">project_id</span> <span class="op">&lt;-</span> <span class="st">"SCPCP000001"</span></span>
<span></span>
<span><span class="co"># Get detailed project metadata</span></span>
<span><span class="va">project_info</span> <span class="op">&lt;-</span> <span class="fu">ScPCAr</span><span class="fu">::</span><span class="fu"><a href="../reference/get_project_info.html">get_project_info</a></span><span class="op">(</span><span class="va">project_id</span><span class="op">)</span></span></code></pre></div>
<p>This returns a <code>list</code> with more detailed information about
the project and the samples within. You can explore the full structure
of this list with <code><a href="https://rdrr.io/r/utils/str.html" class="external-link">str()</a></code> or <code><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">dplyr::glimpse()</a></code>,
but for now we will just look at a few of the components that might be
of interest.</p>
<p>For example, we can look at the set of diagnoses that are present,
and their counts:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">project_info</span><span class="op">$</span><span class="va">diagnoses_counts</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"diagnosis"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                       diagnosis count</span></span>
<span><span class="co">## 1                  Glioblastoma    16</span></span>
<span><span class="co">## 2                 Non-cancerous     1</span></span>
<span><span class="co">## 3             Anaplastic glioma     1</span></span>
<span><span class="co">## 4             High-grade glioma     2</span></span>
<span><span class="co">## 5        Anaplastic astrocytoma     1</span></span>
<span><span class="co">## 6        Diffuse midline glioma     1</span></span>
<span><span class="co">## 7 Pleomorphic xanthoastrocytoma     1</span></span></code></pre>
<p>Similarly, we can look at the data modalities available for this
project, which here includes both single-cell and bulk RNA-seq data:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">project_info</span><span class="op">$</span><span class="va">modalities</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "SINGLE_CELL"  "BULK_RNA_SEQ"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="exploring-sample-information">Exploring sample information<a class="anchor" aria-label="anchor" href="#exploring-sample-information"></a>
</h2>
<div class="section level3">
<h3 id="getting-sample-metadata">Getting sample metadata<a class="anchor" aria-label="anchor" href="#getting-sample-metadata"></a>
</h3>
<p>Let’s look at the samples within our selected project:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get sample information for the project</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu">ScPCAr</span><span class="fu">::</span><span class="fu"><a href="../reference/get_project_samples.html">get_project_samples</a></span><span class="op">(</span><span class="va">project_id</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## # A tibble: 6 × 34</span></span>
<span><span class="co">##   scpca_sample_id scpca_project_id outcome  organism submitter_id participant_id</span></span>
<span><span class="co">##   &lt;chr&gt;           &lt;chr&gt;            &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;        &lt;chr&gt;         </span></span>
<span><span class="co">## 1 SCPCS000001     SCPCP000001      PFS 11 … Homo sa… 834          834           </span></span>
<span><span class="co">## 2 SCPCS000002     SCPCP000001      PFS 11 … Homo sa… 1107         1107          </span></span>
<span><span class="co">## 3 SCPCS000003     SCPCP000001      PFS 7 m… Homo sa… 1115         1115          </span></span>
<span><span class="co">## 4 SCPCS000004     SCPCP000001      PFS 3 m… Homo sa… 1431         1431          </span></span>
<span><span class="co">## 5 SCPCS000005     SCPCP000001      PFS 3 m… Homo sa… 1437         1437          </span></span>
<span><span class="co">## 6 SCPCS000006     SCPCP000001      Stable … Homo sa… 1458         1458          </span></span>
<span><span class="co">## # ℹ 28 more variables: organism_ontology_id &lt;chr&gt;, sex_ontology_term_id &lt;chr&gt;,</span></span>
<span><span class="co">## #   tissue_ontology_term_id &lt;chr&gt;, disease_ontology_term_id &lt;chr&gt;,</span></span>
<span><span class="co">## #   molecular_characteristics &lt;chr&gt;, spinal_leptomeningeal_mets &lt;chr&gt;,</span></span>
<span><span class="co">## #   development_stage_ontology_term_id &lt;chr&gt;,</span></span>
<span><span class="co">## #   self_reported_ethnicity_ontology_term_id &lt;chr&gt;, age &lt;dbl&gt;,</span></span>
<span><span class="co">## #   age_timing &lt;chr&gt;, created_at &lt;dttm&gt;, demux_cell_count_estimate_sum &lt;lgl&gt;,</span></span>
<span><span class="co">## #   diagnosis &lt;chr&gt;, disease_timing &lt;chr&gt;, has_bulk_rna_seq &lt;lgl&gt;, …</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="getting-detailed-sample-information">Getting detailed sample information<a class="anchor" aria-label="anchor" href="#getting-detailed-sample-information"></a>
</h3>
<p>We can also get detailed information for a specific sample:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_id</span> <span class="op">&lt;-</span> <span class="st">"SCPCS000001"</span></span>
<span><span class="co"># Get detailed sample metadata</span></span>
<span><span class="va">sample_detail</span> <span class="op">&lt;-</span> <span class="fu">ScPCAr</span><span class="fu">::</span><span class="fu"><a href="../reference/get_sample_info.html">get_sample_info</a></span><span class="op">(</span><span class="va">sample_id</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">sample_detail</span>, max.level <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 28</span></span>
<span><span class="co">##  $ additional_metadata          :List of 12</span></span>
<span><span class="co">##  $ age                          : chr "14"</span></span>
<span><span class="co">##  $ age_timing                   : chr "diagnosis"</span></span>
<span><span class="co">##  $ computed_files               :'data.frame':   2 obs. of  17 variables:</span></span>
<span><span class="co">##  $ created_at                   : chr "2025-08-26T07:31:52.502575Z"</span></span>
<span><span class="co">##  $ demux_cell_count_estimate_sum: NULL</span></span>
<span><span class="co">##  $ diagnosis                    : chr "Anaplastic glioma"</span></span>
<span><span class="co">##  $ disease_timing               : chr "Initial diagnosis"</span></span>
<span><span class="co">##  $ has_bulk_rna_seq             : logi TRUE</span></span>
<span><span class="co">##  $ has_cite_seq_data            : logi FALSE</span></span>
<span><span class="co">##  $ has_multiplexed_data         : logi FALSE</span></span>
<span><span class="co">##  $ has_single_cell_data         : logi TRUE</span></span>
<span><span class="co">##  $ has_spatial_data             : logi FALSE</span></span>
<span><span class="co">##  $ includes_anndata             : logi TRUE</span></span>
<span><span class="co">##  $ is_cell_line                 : logi FALSE</span></span>
<span><span class="co">##  $ is_xenograft                 : logi FALSE</span></span>
<span><span class="co">##  $ modalities                   : chr [1:2] "SINGLE_CELL" "BULK_RNA_SEQ"</span></span>
<span><span class="co">##  $ multiplexed_with             : list()</span></span>
<span><span class="co">##  $ project                      :List of 37</span></span>
<span><span class="co">##  $ sample_cell_count_estimate   : int 3422</span></span>
<span><span class="co">##  $ scpca_id                     : chr "SCPCS000001"</span></span>
<span><span class="co">##  $ seq_units                    : chr [1:2] "bulk" "cell"</span></span>
<span><span class="co">##  $ sex                          : chr "F"</span></span>
<span><span class="co">##  $ subdiagnosis                 : chr "NA"</span></span>
<span><span class="co">##  $ technologies                 : chr [1:2] "10Xv3" "paired_end"</span></span>
<span><span class="co">##  $ tissue_location              : chr "Right thalamus/midbrain"</span></span>
<span><span class="co">##  $ treatment                    : chr "Debulking, RT, irinotecan/cetuximab"</span></span>
<span><span class="co">##  $ updated_at                   : chr "2025-08-26T07:31:52.502598Z"</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check which data modalities are available for this sample</span></span>
<span><span class="va">sample_detail</span><span class="op">$</span><span class="va">modalities</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "SINGLE_CELL"  "BULK_RNA_SEQ"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="authentication">Authentication<a class="anchor" aria-label="anchor" href="#authentication"></a>
</h2>
<p>To download actual data files, you need an agree to the terms of
service and obtain an authentication token. To view the terms of service
in a web browser, run <code><a href="../reference/get_auth.html">view_terms()</a></code>. Then you can use the
<code><a href="../reference/get_auth.html">get_auth()</a></code> function, providing your email address and
agreeing to the terms of service:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get an authentication token</span></span>
<span><span class="co"># Replace with your actual email address</span></span>
<span><span class="va">my_email</span> <span class="op">&lt;-</span> <span class="st">"your.email@example.com"</span></span>
<span><span class="va">auth_token</span> <span class="op">&lt;-</span> <span class="fu">ScPCAr</span><span class="fu">::</span><span class="fu"><a href="../reference/get_auth.html">get_auth</a></span><span class="op">(</span>email <span class="op">=</span> <span class="va">my_email</span>, agree <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><strong>Important Notes:</strong></p>
<ul>
<li>You must set <code>agree = TRUE</code> to indicate you accept the
terms of service</li>
<li>Replace <code>"your.email@example.com"</code> with your actual email
address</li>
<li>Keep your token secure and don’t share it publicly</li>
</ul>
</div>
<div class="section level2">
<h2 id="downloading-data-for-an-scpca-sample">Downloading data for an ScPCA sample<a class="anchor" aria-label="anchor" href="#downloading-data-for-an-scpca-sample"></a>
</h2>
<div class="section level3">
<h3 id="downloading-singlecellexperiment-objects">Downloading SingleCellExperiment objects<a class="anchor" aria-label="anchor" href="#downloading-singlecellexperiment-objects"></a>
</h3>
<p>Now we can download data for our selected sample. We will use the
<code><a href="../reference/download_sample.html">download_sample()</a></code> function, specifying the sample ID,
authentication token, desired destination directory, and the file
format. The function will download and unpack the files associated with
that sample, and return a list of file paths for the downloaded files
Let’s start with SingleCellExperiment format:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download SingleCellExperiment data for our sample</span></span>
<span><span class="co"># This will create a directory structure under "scpca_data/"</span></span>
<span><span class="va">file_paths</span> <span class="op">&lt;-</span> <span class="fu">ScPCAr</span><span class="fu">::</span><span class="fu"><a href="../reference/download_sample.html">download_sample</a></span><span class="op">(</span></span>
<span>  sample_id <span class="op">=</span> <span class="st">"SCPCS000001"</span>,</span>
<span>  auth_token <span class="op">=</span> <span class="va">auth_token</span>,</span>
<span>  destination <span class="op">=</span> <span class="st">"scpca_data"</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"sce"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Downloading SCPCS000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-10-17.zip ...</span></span></code></pre>
<pre><code><span><span class="co">## Unzipping to scpca_data/SCPCS000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-10-17...</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># List the files that were downloaded</span></span>
<span><span class="va">file_paths</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "scpca_data/SCPCS000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-10-17/README.md"                       </span></span>
<span><span class="co">## [2] "scpca_data/SCPCS000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-10-17/single_cell_metadata.tsv"        </span></span>
<span><span class="co">## [3] "scpca_data/SCPCS000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-10-17/SCPCL000001_celltype-report.html"</span></span>
<span><span class="co">## [4] "scpca_data/SCPCS000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-10-17/SCPCL000001_filtered.rds"        </span></span>
<span><span class="co">## [5] "scpca_data/SCPCS000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-10-17/SCPCL000001_processed.rds"       </span></span>
<span><span class="co">## [6] "scpca_data/SCPCS000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-10-17/SCPCL000001_qc.html"             </span></span>
<span><span class="co">## [7] "scpca_data/SCPCS000001_SINGLE-CELL_SINGLE-CELL-EXPERIMENT_2025-10-17/SCPCL000001_unfiltered.rds"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="understanding-the-downloaded-file-structure">Understanding the downloaded file structure<a class="anchor" aria-label="anchor" href="#understanding-the-downloaded-file-structure"></a>
</h3>
<p>A standard download of SingleCellExperiment data for a sample will
create a directory structure like this:</p>
<pre><code>scpca_data/
└── {sample_id}_{MODALITY}_{FORMAT}_{YYYY-MM-DD}/
    ├── README.md
    ├── single_cell_metadata.tsv
    ├── {library_id}_filtered.rds
    ├── {library_id}_processed.rds
    ├── {library_id}_unfiltered.rds
    ├── {library_id}_qc.html
    └── {library_id}_celltype-report.html</code></pre>
<p>Note that the <code>library_id</code> and <code>sample_id</code>
often have different numbers, as a single sample may have been sequenced
across multiple libraries. For details about the contents of these
files, see the <a href="https://scpca.readthedocs.io/en/latest/download_files.html" class="external-link">ScPCA
Portal documentation</a>.</p>
</div>
<div class="section level3">
<h3 id="loading-data-into-r">Loading data into R<a class="anchor" aria-label="anchor" href="#loading-data-into-r"></a>
</h3>
<p>Now let’s load in the processed data for our sample. First we will
select the processed SingleCellExperiment file from the downloaded
paths, then use <code><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS()</a></code> to load it.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select the processed SCE file using a pattern match for the file name.</span></span>
<span><span class="va">processed_sce_files</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_subset.html" class="external-link">str_subset</a></span><span class="op">(</span><span class="va">file_paths</span>, <span class="st">"_processed\\.rds$"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the first (in this case only) SingleCellExperiment object</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">processed_sce_files</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View a summary of the object</span></span>
<span><span class="va">sce</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SingleCellExperiment </span></span>
<span><span class="co">## dim: 60319 2628 </span></span>
<span><span class="co">## metadata(40): library_id sample_id ... cellassign_reference_version</span></span>
<span><span class="co">##   cellassign_reference_organs</span></span>
<span><span class="co">## assays(3): counts spliced logcounts</span></span>
<span><span class="co">## rownames(60319): ENSG00000223972 ENSG00000243485 ... ENSG00000273496</span></span>
<span><span class="co">##   ENSG00000274175</span></span>
<span><span class="co">## rowData names(4): gene_ids gene_symbol mean detected</span></span>
<span><span class="co">## colnames(2628): GTTCGCTGTTCTCTCG CTCTCAGCATGGATCT ... TCATACTTCTGCGAGC</span></span>
<span><span class="co">##   TCTCACGAGGACGGAG</span></span>
<span><span class="co">## colData names(19): barcodes sum ... consensus_celltype_annotation</span></span>
<span><span class="co">##   consensus_celltype_ontology</span></span>
<span><span class="co">## reducedDimNames(2): PCA UMAP</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="working-with-the-singlecellexperiment-object">Working with the SingleCellExperiment object<a class="anchor" aria-label="anchor" href="#working-with-the-singlecellexperiment-object"></a>
</h3>
<p>Once loaded, you can work with the SingleCellExperiment object using
standard Bioconductor tools. For more information on the contents of the
SingleCellExperiment objects provided by ScPCA, see the <a href="https://scpca.readthedocs.io/en/latest/sce_file_contents.html#components-of-a-singlecellexperiment-object" class="external-link">ScPCA
Portal documentation</a>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Access count data</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 60319  2628</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Access cell metadata</span></span>
<span><span class="va">cell_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cell_metadata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 19 columns</span></span>
<span><span class="co">##                          barcodes       sum  detected subsets_mito_sum</span></span>
<span><span class="co">##                       &lt;character&gt; &lt;numeric&gt; &lt;integer&gt;        &lt;numeric&gt;</span></span>
<span><span class="co">## GTTCGCTGTTCTCTCG GTTCGCTGTTCTCTCG     57012      9097             1494</span></span>
<span><span class="co">## CTCTCAGCATGGATCT CTCTCAGCATGGATCT     52113      8625             4064</span></span>
<span><span class="co">## TCAGCCTCAGGTATGG TCAGCCTCAGGTATGG     49759      8650             5253</span></span>
<span><span class="co">## CCTTGTGGTCCATAGT CCTTGTGGTCCATAGT     58121      9177             3540</span></span>
<span><span class="co">## AGCGCCATCTTCGACC AGCGCCATCTTCGACC     51639      8570             5607</span></span>
<span><span class="co">## CTGTAGATCCATAGGT CTGTAGATCCATAGGT     53179      8962             6712</span></span>
<span><span class="co">##                  subsets_mito_detected subsets_mito_percent     total</span></span>
<span><span class="co">##                              &lt;integer&gt;            &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## GTTCGCTGTTCTCTCG                    13              2.62050     57012</span></span>
<span><span class="co">## CTCTCAGCATGGATCT                    14              7.79844     52113</span></span>
<span><span class="co">## TCAGCCTCAGGTATGG                    14             10.55688     49759</span></span>
<span><span class="co">## CCTTGTGGTCCATAGT                    14              6.09074     58121</span></span>
<span><span class="co">## AGCGCCATCTTCGACC                    14             10.85807     51639</span></span>
<span><span class="co">## CTGTAGATCCATAGGT                    14             12.62152     53179</span></span>
<span><span class="co">##                  prob_compromised miQC_pass scpca_filter sizeFactor  cluster</span></span>
<span><span class="co">##                         &lt;numeric&gt; &lt;logical&gt;  &lt;character&gt;  &lt;numeric&gt; &lt;factor&gt;</span></span>
<span><span class="co">## GTTCGCTGTTCTCTCG      6.09745e-06      TRUE         Keep    5.03806        1</span></span>
<span><span class="co">## CTCTCAGCATGGATCT      8.68181e-06      TRUE         Keep    3.50519        1</span></span>
<span><span class="co">## TCAGCCTCAGGTATGG      2.67246e-06      TRUE         Keep    3.59865        1</span></span>
<span><span class="co">## CCTTGTGGTCCATAGT      7.71935e-07      TRUE         Keep    4.48946        1</span></span>
<span><span class="co">## AGCGCCATCTTCGACC      3.74199e-06      TRUE         Keep    3.38054        1</span></span>
<span><span class="co">## CTGTAGATCCATAGGT      2.00608e-07      TRUE         Keep    3.53366        1</span></span>
<span><span class="co">##                  singler_celltype_ontology singler_celltype_annotation</span></span>
<span><span class="co">##                                &lt;character&gt;                 &lt;character&gt;</span></span>
<span><span class="co">## GTTCGCTGTTCTCTCG                CL:0000540                      neuron</span></span>
<span><span class="co">## CTCTCAGCATGGATCT                CL:0000540                      neuron</span></span>
<span><span class="co">## TCAGCCTCAGGTATGG                CL:0000540                      neuron</span></span>
<span><span class="co">## CCTTGTGGTCCATAGT                CL:0000540                      neuron</span></span>
<span><span class="co">## AGCGCCATCTTCGACC                CL:0000540                      neuron</span></span>
<span><span class="co">## CTGTAGATCCATAGGT                CL:0000540                      neuron</span></span>
<span><span class="co">##                  cellassign_celltype_annotation cellassign_celltype_ontology</span></span>
<span><span class="co">##                                     &lt;character&gt;                  &lt;character&gt;</span></span>
<span><span class="co">## GTTCGCTGTTCTCTCG            Gamma delta T cells                   CL:0000798</span></span>
<span><span class="co">## CTCTCAGCATGGATCT               Purkinje neurons                   CL:0000121</span></span>
<span><span class="co">## TCAGCCTCAGGTATGG               Purkinje neurons                   CL:0000121</span></span>
<span><span class="co">## CCTTGTGGTCCATAGT            Gamma delta T cells                   CL:0000798</span></span>
<span><span class="co">## AGCGCCATCTTCGACC               Purkinje neurons                   CL:0000121</span></span>
<span><span class="co">## CTGTAGATCCATAGGT               Purkinje neurons                   CL:0000121</span></span>
<span><span class="co">##                  cellassign_max_prediction consensus_celltype_annotation</span></span>
<span><span class="co">##                                  &lt;numeric&gt;                   &lt;character&gt;</span></span>
<span><span class="co">## GTTCGCTGTTCTCTCG                  1.000000                       Unknown</span></span>
<span><span class="co">## CTCTCAGCATGGATCT                  1.000000                        neuron</span></span>
<span><span class="co">## TCAGCCTCAGGTATGG                  0.963709                        neuron</span></span>
<span><span class="co">## CCTTGTGGTCCATAGT                  1.000000                       Unknown</span></span>
<span><span class="co">## AGCGCCATCTTCGACC                  1.000000                        neuron</span></span>
<span><span class="co">## CTGTAGATCCATAGGT                  1.000000                        neuron</span></span>
<span><span class="co">##                  consensus_celltype_ontology</span></span>
<span><span class="co">##                                  &lt;character&gt;</span></span>
<span><span class="co">## GTTCGCTGTTCTCTCG                          NA</span></span>
<span><span class="co">## CTCTCAGCATGGATCT                  CL:0000540</span></span>
<span><span class="co">## TCAGCCTCAGGTATGG                  CL:0000540</span></span>
<span><span class="co">## CCTTGTGGTCCATAGT                          NA</span></span>
<span><span class="co">## AGCGCCATCTTCGACC                  CL:0000540</span></span>
<span><span class="co">## CTGTAGATCCATAGGT                  CL:0000540</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Access gene metadata</span></span>
<span><span class="va">gene_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_metadata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 4 columns</span></span>
<span><span class="co">##                        gene_ids gene_symbol        mean  detected</span></span>
<span><span class="co">##                     &lt;character&gt; &lt;character&gt;   &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## ENSG00000223972 ENSG00000223972     DDX11L1 0.000380518 0.0380518</span></span>
<span><span class="co">## ENSG00000243485 ENSG00000243485 MIR1302-2HG 0.000000000 0.0000000</span></span>
<span><span class="co">## ENSG00000284332 ENSG00000284332   MIR1302-2 0.000000000 0.0000000</span></span>
<span><span class="co">## ENSG00000268020 ENSG00000268020      OR4G4P 0.000000000 0.0000000</span></span>
<span><span class="co">## ENSG00000240361 ENSG00000240361     OR4G11P 0.000000000 0.0000000</span></span>
<span><span class="co">## ENSG00000186092 ENSG00000186092       OR4F5 0.000000000 0.0000000</span></span></code></pre>
<p>And of course we can make some standard plots, here a UMAP colored by
the consensus cell type annotation.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View the UMAP, colored by consensus cell type</span></span>
<span><span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plot_reddim.html" class="external-link">plotUMAP</a></span><span class="op">(</span><span class="va">sce</span>, color_by <span class="op">=</span> <span class="st">"consensus_celltype_annotation"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>color <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>title<span class="op">=</span><span class="st">"Cell type"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># remove axis ticks</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    axis.ticks <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="figure/plot-umap-1.png" alt="UMAP of sample SCPCS000001 colored by consensus cell type annotation."><div class="figcaption">UMAP of sample SCPCS000001 colored by consensus
cell type annotation.</div>
</div>
</div>
<div class="section level3">
<h3 id="downloading-data-in-anndatah5ad-format">Downloading data in AnnData/H5AD format<a class="anchor" aria-label="anchor" href="#downloading-data-in-anndatah5ad-format"></a>
</h3>
<p><code>ScPCAr</code> also supports downloading data in AnnData (H5AD)
format, which is commonly used in Python-based single-cell analysis
workflows:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download the same sample in H5AD format</span></span>
<span><span class="va">file_paths_h5ad</span> <span class="op">&lt;-</span> <span class="fu">ScPCAr</span><span class="fu">::</span><span class="fu"><a href="../reference/download_sample.html">download_sample</a></span><span class="op">(</span></span>
<span>  sample_id <span class="op">=</span> <span class="va">sample_id</span>,</span>
<span>  auth_token <span class="op">=</span> <span class="va">auth_token</span>,</span>
<span>  destination <span class="op">=</span> <span class="st">"scpca_data"</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"anndata"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>For more about the expected files when downloading samples in this
format, see the <a href="https://scpca.readthedocs.io/en/stable/download_files.html#anndata-downloads" class="external-link">ScPCA
Portal AnnData docs</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="downloading-projects">Downloading projects<a class="anchor" aria-label="anchor" href="#downloading-projects"></a>
</h2>
<p>In addition to downloading the data for a single sample,
<code>ScPCAr</code> provides a function to download an entire project,
fittingly named <code><a href="../reference/download_project.html">download_project()</a></code>. This function takes a
project id and authentication token as required input, and like the
<code>download_sample</code> function allows you to specify the
destination and format of the downloaded files. There are also a few
other options, such as the ability to download a merged object
containing all samples in the project, and whether to include
multiplexed samples, where multiple samples were pooled and sequenced
together, but have not been demultiplexed. Please see the function
documentation for more information.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download an entire project in SingleCellExperiment format,</span></span>
<span><span class="co"># with separate files for each sample (default behavior).</span></span>
<span><span class="va">project_file_paths</span> <span class="op">&lt;-</span> <span class="fu">ScPCAr</span><span class="fu">::</span><span class="fu"><a href="../reference/download_project.html">download_project</a></span><span class="op">(</span></span>
<span>  project_id <span class="op">=</span> <span class="va">project_id</span>,</span>
<span>  auth_token <span class="op">=</span> <span class="va">auth_token</span>,</span>
<span>  destination <span class="op">=</span> <span class="st">"scpca_data"</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"sce"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Download an entire project in SingleCellExperiment format,</span></span>
<span><span class="co"># with all samples merged into a single object (but not integrated!).</span></span>
<span><span class="va">project_file_paths</span> <span class="op">&lt;-</span> <span class="fu">ScPCAr</span><span class="fu">::</span><span class="fu"><a href="../reference/download_project.html">download_project</a></span><span class="op">(</span></span>
<span>  project_id <span class="op">=</span> <span class="va">project_id</span>,</span>
<span>  auth_token <span class="op">=</span> <span class="va">auth_token</span>,</span>
<span>  destination <span class="op">=</span> <span class="st">"scpca_data"</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"sce"</span>,</span>
<span>  merged <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><summary>
Click to expand R session info
</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">## Running under: macOS Sequoia 15.7.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] SingleCellExperiment_1.28.1 SummarizedExperiment_1.36.0</span></span>
<span><span class="co">##  [3] Biobase_2.66.0              GenomicRanges_1.58.0       </span></span>
<span><span class="co">##  [5] GenomeInfoDb_1.42.3         IRanges_2.40.1             </span></span>
<span><span class="co">##  [7] S4Vectors_0.44.0            BiocGenerics_0.52.0        </span></span>
<span><span class="co">##  [9] MatrixGenerics_1.18.1       matrixStats_1.4.1          </span></span>
<span><span class="co">## [11] ScPCAr_0.1.0                testthat_3.2.3             </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] gridExtra_2.3           httr2_1.2.1             remotes_2.5.0          </span></span>
<span><span class="co">##   [4] rlang_1.1.6             magrittr_2.0.3          scater_1.34.1          </span></span>
<span><span class="co">##   [7] compiler_4.4.3          vctrs_0.6.5             stringr_1.5.1          </span></span>
<span><span class="co">##  [10] profvis_0.4.0           pkgconfig_2.0.3         crayon_1.5.3           </span></span>
<span><span class="co">##  [13] fastmap_1.2.0           XVector_0.46.0          ellipsis_0.3.2         </span></span>
<span><span class="co">##  [16] labeling_0.4.3          scuttle_1.16.0          utf8_1.2.5             </span></span>
<span><span class="co">##  [19] promises_1.3.0          sessioninfo_1.2.2       tzdb_0.5.0             </span></span>
<span><span class="co">##  [22] UCSC.utils_1.2.0        ggbeeswarm_0.7.2        purrr_1.0.2            </span></span>
<span><span class="co">##  [25] xfun_0.52               zlibbioc_1.52.0         cachem_1.1.0           </span></span>
<span><span class="co">##  [28] beachmat_2.22.0         jsonlite_2.0.0          later_1.3.2            </span></span>
<span><span class="co">##  [31] DelayedArray_0.32.0     BiocParallel_1.40.2     irlba_2.3.5.1          </span></span>
<span><span class="co">##  [34] parallel_4.4.3          R6_2.6.1                stringi_1.8.7          </span></span>
<span><span class="co">##  [37] pkgload_1.4.0           brio_1.1.5              Rcpp_1.0.13            </span></span>
<span><span class="co">##  [40] knitr_1.50              usethis_3.1.0           readr_2.1.5            </span></span>
<span><span class="co">##  [43] httpuv_1.6.15           Matrix_1.7-0            tidyselect_1.2.1       </span></span>
<span><span class="co">##  [46] viridis_0.6.5           rstudioapi_0.17.1       abind_1.4-8            </span></span>
<span><span class="co">##  [49] yaml_2.3.10             codetools_0.2-20        miniUI_0.1.1.1         </span></span>
<span><span class="co">##  [52] curl_7.0.0              pkgbuild_1.4.6          lattice_0.22-6         </span></span>
<span><span class="co">##  [55] tibble_3.2.1            shiny_1.9.1             withr_3.0.2            </span></span>
<span><span class="co">##  [58] evaluate_1.0.3          desc_1.4.3              urlchecker_1.0.1       </span></span>
<span><span class="co">##  [61] pillar_1.10.2           generics_0.1.4          rprojroot_2.0.4        </span></span>
<span><span class="co">##  [64] hms_1.1.3               ggplot2_3.5.1           munsell_0.5.1          </span></span>
<span><span class="co">##  [67] scales_1.3.0            xtable_1.8-4            glue_1.8.0             </span></span>
<span><span class="co">##  [70] tools_4.4.3             BiocNeighbors_2.0.1     ScaledMatrix_1.14.0    </span></span>
<span><span class="co">##  [73] fs_1.6.6                cowplot_1.1.3           grid_4.4.3             </span></span>
<span><span class="co">##  [76] tidyr_1.3.1             devtools_2.4.5          colorspace_2.1-1       </span></span>
<span><span class="co">##  [79] GenomeInfoDbData_1.2.13 beeswarm_0.4.0          BiocSingular_1.22.0    </span></span>
<span><span class="co">##  [82] vipor_0.4.7             cli_3.6.5               rsvd_1.0.5             </span></span>
<span><span class="co">##  [85] rappdirs_0.3.3          viridisLite_0.4.2       S4Arrays_1.6.0         </span></span>
<span><span class="co">##  [88] dplyr_1.1.4             gtable_0.3.5            digest_0.6.37          </span></span>
<span><span class="co">##  [91] ggrepel_0.9.6           SparseArray_1.6.2       farver_2.1.2           </span></span>
<span><span class="co">##  [94] htmlwidgets_1.6.4       memoise_2.0.1           htmltools_0.5.8.1      </span></span>
<span><span class="co">##  [97] lifecycle_1.0.4         httr_1.4.7              here_1.0.1             </span></span>
<span><span class="co">## [100] mime_0.13</span></span></code></pre>
</details>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Joshua A. Shapiro, Stephanie J. Spielman, Alex’s Lemonade Stand Foundation.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
